internal/xds/bootstrap/bootstrap.go

```diff
@@ -223,6 +226,12 @@ func (sc *ServerConfig) CredsDialOption() grpc.DialOption {
 	return sc.credsDialOption
 }
 
+// DialerOption returns the first supported Dialer function that specifies how
+// to dial the xDS server from the configuration, as a dial option.
+func (sc *ServerConfig) DialerOption() grpc.DialOption {
+	return sc.dialerOption
+}
```

Comment:
Hmm ... I'm thinking if we should just have a method to return a slice of dial options from this type:

```golang
func (sc *ServerConfig) DialOptions() []grpc.DialOption {
	// return a slice of dial options that contains the creds dial option and optionally one for the dialer
}
```

With this approach, the transport can be oblivious to the dial options, and tomorrow if another dial option needs to be added here, then the transport won't have to change.

Reply:
Acknowledged.  
I think this is a good abstraction, although it might slightly reduce readability and intent of what each dial option does (only two dial options *passed* to the transport).

I think this abstraction would be appropriate in a follow-up PR as it would involve removing the existing `func (sc *ServerConfig) CredsDialOption()`, introduce the new API above, and refactoring the transport, which is unrelated to the current PR of adding a custom Dialer used by the xDS transport and it would reduce this PR's focus/intent/readability.

Reply:
Ack. Would you mind adding a TODO here with the issue that I've filed to track this: https://github.com/grpc/grpc-go/issues/7661.

Thanks.

Reply:
Done.  
Added TODO with reference to https://github.com/grpc/grpc-go/issues/7661 and added a brief explanation.

FYI: I examined the repo and there seems to be different TODO styles:
*  `TODO(username):`
* `TODO(i/issue-number):`
* `TODO(https://github.com/grpc/grpc-go/issues/issue-number):`
* `TODO:`
* etc.

Generally it is recommended to link a issue/bug rather than a person, thus I chose `TODO(https://github.com/grpc/grpc-go/issues/issue-number)` to be less ambiguous, this also makes it easy for users to click to link directly within IDE.